### PR TITLE
Add Windows ARM64 support for libyara .NET wrappers.

### PR DESCRIPTION
**Pull request by HydrophobicMinghao**
1 Add ARM64 build platform to the solution.
2 Modify the version of native libyara package to a version that has correct ARM64 static lib and build files.
3 Align the target framework version of project and nuspec to net462 for `Microsoft.O365.Security.Native.libyara.NET`.
4 Add and fix content of docs and notes related to ARM64 support.

### CODE REVIEW

File: Microsoft.O365.Security.Native.libyara.NET.Core.nuspec [hunk 1]
```diff
@@ -2,33 +2,58 @@
 <package >
   <metadata>
     <id>Microsoft.O365.Security.Native.libyara.NET.Core</id>
-    <version>4.2.3</version>
+    <version>4.3.0</version>
```
Comment (henriblMSFT) [inline Microsoft.O365.Security.Native.libyara.NET.Core.nuspec]: Let's align the package version to the version of libyara so 4.5.1

File: Microsoft.O365.Security.Native.libyara.NET.nuspec [hunk 2]
```diff
@@ -1,17 +1,18 @@
 <?xml version="1.0"?>
-<package >
+<package>
   <metadata>
     <id>Microsoft.O365.Security.Native.libyara.NET</id>
-    <version>4.2.3</version>
+    <version>4.3.0</version>
```
Comment (henriblMSFT) [inline Microsoft.O365.Security.Native.libyara.NET.nuspec]: 4.5.1

**Review by henriblMSFT — CHANGES_REQUESTED** [review]

File: Microsoft.O365.Security.Native.libyara.NET.Core.nuspec (hunk 1 above)
Comment (HydrophobicMinghao) [reply Microsoft.O365.Security.Native.libyara.NET.Core.nuspec]: Updated the package version to 4.5.1.

File: Microsoft.O365.Security.Native.libyara.NET.nuspec (hunk 2 above)
Comment (HydrophobicMinghao) [reply Microsoft.O365.Security.Native.libyara.NET.nuspec]: Updated the package version to 4.5.1.

**Review by henriblMSFT — APPROVED** [review]

**Pull request merged**